client/components/main/index.jsx

```diff
@@ -17,3 +29,44 @@ module.exports = React.createClass( {
 		);
 	}
 } );
+
+/**
+ * As apparent below, the idea is simple: connect Main to Redux so it knows
+ * whether it should render a guided tour rather than `primary`.
+ *
```

Comment:
A chat with @ehg lead to the quick conclusion that, while the general idea likely makes sense, `client/layout` is a much more suitable component than `Main`, so I'll get to that.